File: src/backend/Application/GoalAggregate/Commands/EditGoalAndActionItems/EditGoalAndActionItemsCommand.cs
using InternetSafetyPlan.Application.Base;
using InternetSafetyPlan.Domain.GoalAggregate;
using MediatR;

namespace InternetSafetyPlan.Application.GoalAggregate.Commands;

public record EditGoalAndActionItemsCommand(Guid Id, Guid UltimateGoalId, string Name, GoalCategory Category, string? Description = null, DateTime? DueDate = null, List<EditGoalAndActionItemsCommand_ActionItem>? ActionItems = null) : ICommand<Unit>;
public record EditGoalAndActionItemsCommand_ActionItem(Guid Id, string Name, string? Description = null, DateTime? DueDate = null);

File: src/frontend/src/library/goalAggregate/commands/EditGoalAndActionItemsCommand.ts
export class EditGoalAndActionItemsCommand {
  public id: string;
  public ultimateGoalId: string;
  public name: string;
  public category: number;
  public description: string | null;
  public dueDate: string | null;
  public actionItems: EditGoalAndActionItemsCommand_ActionItem[];

  constructor(id: string, ultimateGoalId: string, name: string, category: number, description: string | null, dueDate: string | null, actionItems: EditGoalAndActionItemsCommand_ActionItem[]) {
    this.id = id;
    this.ultimateGoalId = ultimateGoalId;
    this.name = name;
    this.category = category;
    this.description = description;
    this.dueDate = dueDate;
    this.actionItems = actionItems;
  }
}

export class EditGoalAndActionItemsCommand_ActionItem {
  public id: string;
  public name: string;
  public description: string | null;
  public dueDate: string;

  constructor(id: string, name: string, description: string | null, dueDate: string) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.dueDate = dueDate;
  }
}